src/fixtures/logstash_fields.js

```diff
@@ -1,36 +1,65 @@
-    { name: 'script murmur3',     type: 'murmur3',    scripted: true, script: '1234', lang: 'expression'},
-  ].map(function (field) {
-    field.count = field.count || 0;
-    field.scripted = field.scripted || false;
-    return field;
-  });
+  return [
+    //                                  |indexed
+    //                                  |      |analyzed
+    //                                  |      |      |aggregatable
+    //                                  |      |      |      |searchable
+    // name                type         |      |      |      |     |metadata
+    ['bytes',             'number',     true,  true,  true,  true,  { count: 10 } ],
+    ['ssl',               'boolean',    true,  true,  true,  true,  { count: 20 } ],
+    ['@timestamp',        'date',       true,  true,  true,  true,  { count: 30 } ],
+    ['time',              'date',       true,  true,  true,  true,  { count: 30 } ],
+    ['@tags',             'string',     true,  true,  true,  true ],
+    ['utc_time',          'date',       true,  true,  true,  true ],
+    ['phpmemory',         'number',     true,  true,  true,  true ],
+    ['ip',                'ip',         true,  true,  true,  true ],
+    ['request_body',      'attachment', true,  true,  true,  true ],
+    ['point',             'geo_point',  true,  true,  true,  true ],
+    ['area',              'geo_shape',  true,  true,  true,  true ],
+    ['hashed',            'murmur3',    true,  true,  false, true ],
+    ['geo.coordinates',   'geo_point',  true,  true,  true,  true ],
+    ['extension',         'string',     true,  true,  true,  true ],
+    ['machine.os',        'string',     true,  true,  true,  true ],
+    ['geo.src',           'string',     true,  true,  true,  true ],
+    ['_id',               'string',     false, false, true,  true ],
+    ['_type',             'string',     false, false, true,  true ],
+    ['_source',           'string',     false, false, true,  true ],
+    ['custom_user_field', 'conflict',   false, false, true,  true ],
+    ['script string',     'string',     false, false, true,  false, { script: '\'i am a string\'' } ],
+    ['script number',     'number',     false, false, true,  false, { script: '1234' } ],
+    ['script date',       'date',       false, false, true,  false, { script: '1234', lang: 'painless' } ],
+    ['script murmur3',    'murmur3',    false, false, true,  false, { script: '1234' } ],
+  ].map(function (row) {
+    const [
+      name,
+      type,
+      indexed,
+      analyzed,
+      aggregatable,
+      searchable,
+      metadata = {}
+    ] = row;
+
+    const {
+      count = 0,
+      script,
+      lang = script ? 'expression' : undefined,
+      scripted = !!script,
+    } = metadata;
 
-  return sourceData;
+    return {
+      name,
+      type,
+      indexed,
+      analyzed,
+      aggregatable,
+      searchable,
+      count,
+      script,
+      lang,
+      scripted,
+    };
+  });
```

Comment:
I think I am missing what changed here. Is this a rewrite of the original? I guess 'expression' is now no longer hardcoded.